mozjs/mozglue/misc/Mutex_posix.cpp

```diff
@@ -66,11 +66,9 @@ mozilla::detail::MutexImpl::MutexImpl() {
                     "mozilla::detail::MutexImpl::MutexImpl: "
                     "pthread_mutexattr_settype failed");
 #  elif defined(POLICY_KIND)
-  if (__builtin_available(macOS 10.14, *)) {
-    TRY_CALL_PTHREADS(pthread_mutexattr_setpolicy_np(&attr, POLICY_KIND),
-                      "mozilla::detail::MutexImpl::MutexImpl: "
-                      "pthread_mutexattr_setpolicy_np failed");
-  }
+  (pthread_mutexattr_setpolicy_np(&attr, POLICY_KIND),
```

Comment:
Can you elaborate on why this patch is needed?

Reply:
__builtin_available causes linking errors when linked statically, and I'm not sure how to work around that. The mutex policy is invalid on earlier versions of macOS, like the ones we run our CI on, but that just returns an error code but is otherwise ignored. That means we can try applying the policy and ignore it if it fails.